packages/blocks/src/page-block/default/selection-manager.ts

```diff
@@ -379,42 +354,34 @@ export class DefaultSelectionManager {
     this.state.clear();
     this._signals.updateSelectedRects.emit([]);
     this._signals.updateEmbedRects.emit([]);
-    if ((e.raw.target as HTMLElement).tagName === 'DEBUG-MENU') return;
 
-    const dividerBlockComponent = (e.raw.target as HTMLElement).closest(
-      'divider-block'
-    ) as HTMLElement;
-    const clickBlockInfo = getHoverBlockOptionByPosition(
+    if ((e.raw.target as HTMLElement).tagName === 'DEBUG-MENU') return;
+    const clickBlockInfo = getBlockOptionByPosition(
       this._blocks,
       e.raw.pageX,
       e.raw.pageY
     );
-    if (clickBlockInfo?.model.type === 'image') {
+    if (clickBlockInfo && clickBlockInfo.model) {
+      const { model } = clickBlockInfo;
+      const page = getDefaultPageBlock(model);
+      page.lastSelectionPosition = 'start';
+    }
+    if (clickBlockInfo && NON_TEXT_ARR.includes(clickBlockInfo.model.type)) {
```

Comment:
Please add a `matchTypes` in the same module of `matchFlavours`, so as to replace `NON_TEXT_ARR.includes` (and other similar places) with them, this benefits static analysis about the usage of block flavours and sub-types.